lib/ui/painting.dart

```diff
@@ -4259,6 +4268,124 @@ class FragmentProgram extends NativeFieldWrapperClass1 {
+/// allocating and re-initializing the uniform buffer and samplers. However,
+/// if two [FragmentShader] objects with different float uniforms or samplers
+/// are required to exist simultaneously, they must be obtained from two
+/// different calls to the [FragmentProgram]'s `fragmentShader()` method.
+///
+/// Consider a fragment shader with uniforms like the following:
+///
+/// ```glsl
+/// layout (location = 0) uniform float a;
+/// layout (location = 1) uniform vec2 b;
+/// layout (location = 2) uniform vec3 c;
+/// layout (location = 3) uniform mat2x2 d;
+/// layout (location = 4) uniform sampler2d s;
+/// ```
+///
+/// If this shader is loaded into a [FragmentProgram] object `program`, a
+/// [FragmentShader] object can be constructed as follows:
+/// ```dart
+/// ImageShader sampler;
+/// final FragmentShader shader = program.fragmentShader()
+/// ..floatUniforms[0] = 0.0
+/// ..floatUniforms[1] = 1.0
+/// ..floatUniforms[2] = 2.0
+/// ..floatUniforms[3] = 3.0
+/// ..floatUniforms[4] = 4.0
+/// ..floatUniforms[5] = 5.0
+/// ..floatUniforms[6] = 6.0
+/// ..floatUniforms[7] = 7.0
+/// ..floatUniforms[8] = 8.8
+/// ..floatUniforms[9] = 9.0
+/// ..setSampler(0, sampler);
+///
+/// Paint paint = Paint()..shader = shader;
+/// ```
+///
+/// The uniforms will then be set as follows:
+///
+/// a: 0.0
+/// b: [1.0, 2.0]
+/// c: [3.0, 4.0, 5.0]
+/// d: [6.0, 7.0, 8.0, 9.0] // 2x2 matrix in column-major order
+///
+/// On subsequent frames, if only the uniform `a` is required to vary, then
+/// only `shader.floatUniforms[0]` need be modified. The sampler uniforms will
+/// also persist in the same way.
+class FragmentShader extends Shader {
+  FragmentShader._(this._program) : super._() {
+    _floats = _constructor(
+      _program,
+      _program._uniformFloatCount,
+      _program._samplerCount,
+    );
+  }
+
+  /// The [FragmentProgram] instance that generated this [FragmentShader].
+  final FragmentProgram _program;
+
+  late final Float32List _floats;
+
+  /// The data used to initialize the float uniforms of the shader program.
+  Float32List get floatUniforms {
+    assert(!debugDisposed, 'Tried to accesss floats on a disposed Shader: $this');
+    return _floats;
+  }
+
+  /// Sets the sampler uniform at `index`.
+  ///
+  /// All the sampler uniforms that a shader expects must be provided or the
+  /// results will be undefined.
+  void setSampler(int index, ImageShader sampler) {
+    assert(!debugDisposed, 'Tried to access samplers on a disposed Shader: $this');
+    _setSampler(index, sampler);
+  }
+
+  @override
+  bool operator ==(Object other) {
+    if (identical(this, other)) {
+      return true;
+    }
+    if (other.runtimeType != runtimeType) {
+      return false;
+    }
+    return other is FragmentShader
+        && other._program == _program
+        && _listEquals<double>(other._floats, _floats);
+  }
+
+  @override
+  int get hashCode => Object.hash(
+    _program,
+    Object.hashAll(_floats),
+  );
+
```

Comment:
Since this object is mutable, it shouldn't have equality/hashcode that is content aware.

Reply:
Thanks! Removed.